internal/tracker/status_tracker.go

```diff
@@ -0,0 +1,527 @@
+// Package tracker provides status tracking functionality for ModelValidation resources
+package tracker
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sort"
+	"sync"
+	"time"
+
+	"golang.org/x/time/rate"
+
+	"github.com/sigstore/model-validation-operator/api/v1alpha1"
+	"github.com/sigstore/model-validation-operator/internal/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// StatusTrackerImpl tracks injected pods and namespaces for ModelValidation resources
+type StatusTrackerImpl struct {
```

Comment:
I'm having a hard time following the logic of the StatusTracker.

Could you clarify how the state of the tracked objects is preserved or recreated? I'm concerned about what happens to this information when the operator pod is restarted caused by upgrade, or moved to a different node.

Reply:
The status tracker relies on the two controllers to provide the information, one is for tracking the pods and the other for tracking the ModelValidations.  If the operator is restarted then the controllers will go through the List/Watch cycle.  I do have a seeding part for the pods, but this is not strictly necessary as the pod controller will eventually provide the same information.